physutils/physio.py

```diff
@@ -0,0 +1,534 @@
+
+    Parameters
+    ----------
+    exclude : list, optional
+        What arguments to exclude from provided argument : value dictionary.
+        Default: ['data']
+    serializable : bool, optional
+        Whether to coerce argument values to JSON serializable form. Default:
+        True
+
+    Returns
+    -------
+    function: str
+        Name of calling function
+    provided : dict
+        Dictionary of function arguments and provided values
+    """
+
+    exclude = ["data"] if exclude is None else exclude
+    if not isinstance(exclude, list):
+        exclude = [exclude]
+
+    # get one function call up the stack (the bottom is _this_ function)
+    calling = inspect.stack(0)[1]
+    frame, function = calling.frame, calling.function
+
+    # get all the args / kwargs from the calling function
+    argspec = inspect.getfullargspec(frame.f_globals[function])
+    args = sorted(argspec.args + argspec.kwonlyargs)
+
+    # save arguments + argument values for everything not in `exclude`
+    provided = {k: frame.f_locals[k] for k in args if k not in exclude}
+
+    # if we want `provided` to be serializable, we can do a little cleaning up
+    # this is NOT foolproof, but will coerce numpy arrays to lists which tends
+    # to be the main issue with these sorts of things
+    if serializable:
+        for k, v in provided.items():
+            if hasattr(v, "tolist"):
+                provided[k] = v.tolist()
+
+    function = inspect.getmodule(frame).__name__ + "." + function
+
+    return function, provided
+
+
+def check_physio(data, ensure_fs=True, copy=False):
+    """
+    Checks that `data` is in correct format (i.e., `peakdet.Physio`)
+
+    Parameters
+    ----------
+    data : Physio_like
+    ensure_fs : bool, optional
+        Raise ValueError if `data` does not have a valid sampling rate
+        attribute.
+    copy: bool, optional
+        Whether to return a copy of the provided data. Default: False
+
+    Returns
+    -------
+    data : peakdet.Physio
+        Loaded physio object
+
+    Raises
+    ------
+    ValueError
+        If `ensure_fs` is set and `data` doesn't have valid sampling rate
+    """
+
+    from physutils.io import load_physio
+
+    if not isinstance(data, Physio):
+        data = load_physio(data)
+    if ensure_fs and np.isnan(data.fs):
+        raise ValueError("Provided data does not have valid sampling rate.")
+    if copy is True:
+        return new_physio_like(
+            data,
+            data.data,
+            copy_history=True,
+            copy_metadata=True,
+            copy_suppdata=True,
+            copy_label=True,
+            copy_physio_type=True,
+            copy_computed_metrics=True,
+        )
+    return data
+
+
+def new_physio_like(
+    ref_physio,
+    data,
+    *,
+    fs=None,
+    suppdata=None,
+    dtype=None,
+    copy_history=True,
+    copy_metadata=True,
+    copy_suppdata=True,
+    copy_label=True,
```

Comment:
`copy_label`, `copy_physio_type` and `copy_computed_metrics` don't seem to be used in that function

Reply:
fixed that as well